### PR TITLE
Scope and rename assertion/error checking macross

### PR DESCRIPTION
**Pull request by sethrj**
See https://github.com/celeritas-project/celeritas/discussions/97, our slack vote, and our resolution of the discussion in today's meeting. I ported a utility script from Exnihilo to make the renaming operation  simple and reproducible in case there are outdated branches with error checking.

### CODE REVIEW

File: src/base/Assert.hh [hunk 1]
```diff
@@ -102,33 +104,33 @@
...
 
 #ifndef __CUDA_ARCH__
-#    define INSIST(COND, MSG) CELER_RUNTIME_ASSERT_(COND, MSG)
+#    define CELER_VALIDATE(COND, MSG) CELER_RUNTIME_ASSERT_(COND, MSG)
 #    define CELER_NOT_CONFIGURED(WHAT) CELER_DEBUG_FAIL_(WHAT, unconfigured)
 #else
-#    define INSIST(COND, MSG)                                                  \
+#    define CELER_VALIDATE(COND, MSG)                                          \
         ::celeritas::throw_debug_error(::celeritas::DebugErrorType::assertion, \
                                        "Insist cannot be called from device "  \
```
Comment (pcanal) [inline src/base/Assert.hh:129]:
```suggestion
                                       "CELER_VALIDATE cannot be called from device "  \
```

File: src/base/Assert.hh [hunk 2]
```diff
@@ -102,33 +104,33 @@
...
-                                       "code",                                 \
+                                       "from device code",                     \
```
Comment (pcanal) [inline src/base/Assert.hh:130]: Isn't this string concatenate with the previous string and thus the message now reads ".... from device from device code ...." ?

File: src/base/Macros.hh [hunk 3]
```diff
@@ -94,8 +94,8 @@
  * or https://msdn.microsoft.com/en-us/library/1b3fsfxw.aspx=
  *
  * \note This macro is usually unused; instead, the macro \c
- * CHECK_UNREACHABLE defined in DBC.hh should be used instead (to provide
- * a more detailed error message in case the point *is* reached).
+ * CELER_ASSERT_UNREACHABLE() defined in DBC.hh should be used instead (to
```
Comment (pcanal) [inline src/base/Macros.hh:97]: Did you mean:
```suggestion
 * CELER_ASSERT_UNREACHABLE() defined in base/Assert.hh should be used instead (to
```
?

File: src/base/Assert.hh (hunk 1 above)
Comment (sethrj) [reply src/base/Assert.hh:129]: Whoops I thought I pushed that fix.

File: src/base/Macros.hh (hunk 3 above)
Comment (sethrj) [reply src/base/Macros.hh:97]: Hah, nice catch -- a holdover from the file in Shift in which I originally wrote the macro 😅

File: src/base/Assert.hh (hunk 2 above)
Comment (sethrj) [reply src/base/Assert.hh:130]: Yeah, something messed up perhaps in my inline invocation of clang-format? I'd replaced the macro which reformatted the line thusly, then the line with "Insist" reverted somehow.

**Review by pcanal — APPROVED** [review]

**Pull request merged**